src/proof_rules/induction.rs

```diff
@@ -207,17 +202,17 @@ impl Encoding for KIndAnnotation {
         resolve.visit_expr(invariant)
     }
 
-    fn check_calculus(&self, calculus: &Calculus, direction: Direction) -> Result<(), ()> {
+    fn is_calculus_allowed(&self, calculus: &Calculus, direction: Direction) -> bool {
         if direction
```

Comment:
The logic here seems a bit convoluted. Why not simply immediately return `direction ==  match calculus.calculus_type {
                CalculusType::Wp | CalculusType::Ert => Direction::Up,
                CalculusType::Wlp => Direction::Down,
            }`?